### ISSUE TITLE
Migrate `match_and_detrend`

### ISSUE THREAD

**Issue by jbusecke** [jbusecke/cmip6_omz#17]
We should for now migrate [this](https://github.com/jbusecke/aguadv_omz_busecke_2021/blob/0ce9ec69295cfa77623723083913e60022dada50/aguadv_omz_busecke_2021/cmip6_stash.py#L211-L265) to the `upstream_stash.py` module.

You should replace the call to `detrend` with the new [`remove_drift`](https://github.com/jbusecke/cmip6_preprocessing/blob/7e163f9716bbe7b055339f523060338053aa6339/cmip6_preprocessing/drift_removal.py#L312) from cmip6_preprocessing.

**Comment by jbusecke** [comment]
I think most of this code:
```python 
trend_ds = trend_dict[match_trend_names[0]]
                # !!!!!! I need this in the regression files otherwise it might not be 100% accurate.
                # But I think for most runs here 1850 is right... !!!!
                ref_date = construct_cfdate(
                    0, "hours since 1850-1-15", _get_calendar(ds.time)
                )
                #                 # i might want to preserve the attrs in the detrend function?
                #                 attrs = ds[ds.attrs["variable_id"]].attrs
                da_detrended = detrend(
                    ds[ds.attrs["variable_id"]], trend_ds["slope"], ref_date=ref_date
                )
                #                 da_detrended.attrs = attrs
                da_detrended.attrs.update(
                    {"detrended_with_file": trend_ds.attrs["filepath"]}
                )
                ds[ds.attrs["variable_id"]] = da_detrended.reset_coords(drop=True)
                data_dict_detrended[name] = ds
```

can also be eliminated

```python
trend_ds = trend_dict[match_trend_names[0]]
                da_detrended = detrend(
                    ds[ds.attrs["variable_id"]], trend_ds["slope"], ref_date=ref_date
                ) # might need some minor syntax adjustment
                ds[ds.attrs["variable_id"]] = da_detrended.reset_coords(drop=True)
                data_dict_detrended[name] = ds
```

**Comment by sditkovsky** [comment]
Got the `match_and_detrend` function working today in a notebook with relatively few hiccups. There were a couple minor clashes with the `remove_trend()` function:

- the trend files in .../linear_regression_time_zarr_multimember/ use 'slope' as the variable name, but `remove_trend()` expects the slope variable to be an actual name like 'o2', 'so', etc. Easy fix to just rename things in the function, but it does feel like a clash of conventions to be aware of. 
- Getting an error from [line 350](https://github.com/jbusecke/cmip6_preprocessing/blob/c8df6fde13319642f1a5a4b26f3be5e5fd1f4fe1/cmip6_preprocessing/drift_removal.py#L350) in cmip6_pp.drift_removal. It complains that Array has to attribute tolist(). Not sure why this is happening because the same sort of thing works elsewhere. Was able to get around this for today, though, by dropping `trend_time_range` from dims. 

Still working through the last (sigma-binning) step of the pipeline, and ignoring the nesting step for now (just merging manually for CanESM5). I'll bring `match_and_detrend` to `upstream_stash` soon

**Comment by jbusecke** [comment]
Yeah that seems fine to me. Eventually we can overhaul that last step with brand new trend files and some more functionality in cmip6_pp. I would just double check that the detrending is working properly (e.g. print some random point from both the detrended and non-detrended member).

**Comment by jbusecke** [comment]
Closed via #19 

FYI @sditkovsky, if you put this line (`- [x] Closes #xx` with xx representing an issue number) in a PR, it will close that one automatically once the PR is merged!
